Source/stores.cpp

```diff
@@ -183,26 +180,26 @@ void PrintStoreItem(ItemStruct *x, int l, uint16_t flags)
 		strcpy(sstr, fmt::format(_("Damage: {:d}-{:d}  "), x->_iMinDam, x->_iMaxDam).c_str());
 	if (x->_iClass == ICLASS_ARMOR)
 		strcpy(sstr, fmt::format(_("Armor: {:d}  "), x->_iAC).c_str());
-	if (x->_iMaxDur != DUR_INDESTRUCTIBLE && x->_iMaxDur) {
+	if (x->_iMaxDur != DUR_INDESTRUCTIBLE && x->_iMaxDur != 0) {
 		strcpy(tempstr, fmt::format(_("Dur: {:d}/{:d},  "), x->_iDurability, x->_iMaxDur).c_str());
 		strcat(sstr, tempstr);
 	} else {
 		strcat(sstr, _("Indestructible,  "));
 	}
 	if (x->_itype == ITYPE_MISC)
 		sstr[0] = '\0';
-	str = x->_iMinStr;
-	mag = x->_iMinMag;
-	dex = x->_iMinDex;
+	int8_t str = x->_iMinStr;
+	uint8_t mag = x->_iMinMag;
+	int8_t dex = x->_iMinDex;

```

Comment:
Is there an actual reason for the mixed types?

Reply:
magic can exceed 128, afaik it's not safe to simply change the two others to unsigned.

Reply:
Scary